hardware_isolation_manager.cpp

```diff
@@ -397,4 +397,49 @@ void Manager::restore()
         this->createEntryForRecord(record);
     });
 }
+
+void Manager::handleHostIsolatedHardwares()
+{
+    // Wait for some time to get the final isolated hardware record list
+    // which are updated by the host because of the atomicity on partition
+    // file (which is used to isolated hardware details) between BMC and Host.
+    sleep(5);
+
+    openpower_guard::GuardRecords records = openpower_guard::getAll();
+
+    std::for_each(records.begin(), records.end(), [this](const auto& record) {
+        // Skipping fake record (GARD_Reconfig) because, fake record is
+        // created for internal purposes to use by BMC and Hostboot.
+        if (record.errType == openpower_guard::GardType::GARD_Reconfig)
+        {
+            return;
+        }
+
+        // Make sure the isolated hardware record is resolved (recordId will
+        // get change as "0xFFFFFFFF") by host OR the record is already exist.
+        auto isolatedHwIt = std::find_if(
```

Comment:
I am not clear about this
are you looking at whether there is a 0xFFFF... or a valid record id for each guarded unit? there can be multiple entries for same target right?

Reply:
```
 return (
                    (isolatedHw.second->getEntityPath() == record.targetId) &&
                    ((record.recordId == 0xFFFFFFFF) ||
                     (isolatedHw.second->getEntryRecId() == record.recordId)));
```

- First will make sure targetId (aka EntityPath of isolated hardware) is present `(isolatedHw.second->getEntityPath() == record.targetId)` in BMC isolated entries **AND** is it resolved `(record.recordId == 0xFFFFFFFF)` by Host **OR** the existing record `(isolatedHw.second->getEntryRecId() == record.recordId)` in BMC entires  So, from this we will get the BMC entry of the record which is readed from partition and it will make sure if it is resolved by host then it will update the respective bmc entry as resolved (https://github.com/ibm-openbmc/openpower-hw-isolation/pull/2/commits/1cec41d7420a64ad790d658a17a4401da5082000#diff-5263f98e5cdba1f39332153cad7bedef71d58e87b9a39a2681cb297e095438d0R438). If not resolved nothing will do and go for next record from partition.
- If the record which is readed, is not found in BMC entries then it will assume the new hardware is isolated by Host so BMC need to create a new entry (https://github.com/ibm-openbmc/openpower-hw-isolation/pull/2/commits/1cec41d7420a64ad790d658a17a4401da5082000#diff-5263f98e5cdba1f39332153cad7bedef71d58e87b9a39a2681cb297e095438d0R434)